BriefAssistant/ClientApp/app/components/form/form.component.ts

```diff
@@ -0,0 +1,289 @@
+    judgeLastName: string;
+    opponentFirstName: string;
+    opponentLastName: string;
+
+
+
+}
+interface Address {
+    street: string;
+    street2: string;
+    city: string;
+    state: State;
+    zip: string;
+
+}
+@NgModule({
+    imports: [
+        BrowserModule,
+        HttpModule
+    ]
+
+})
+@Component({
+    selector: 'dataForm',
+    templateUrl: './form.component.html',
+   // viewProviders: [HttpModule],
+    styleUrls: ['./form.component.css']
+})
+export class FormComponent {
+    briefInfo: BriefInfo;
+    appellant: Appellant;
+    address: Address;
+    circuitCourtCase: CircuitCourtCase;
+    ngOnInit() {
+        //initialize form
+
+        this.appellant = {
+            firstName: "",
+            lastName: "",
+            email:"",
+            phone: ""
+
+        }
+        this.address= {
+            street: "",
+            street2:"",
+            city:"",
+            state: State.WI,
+            zip:"",
+
+        }
+        this.circuitCourtCase = {
+            county: County.Milwaukee,
+            caseNumber:'',
+            role: Role.Defendent,
+            judgeFirstName:'',
+            judgeLastName: '',
+            opponentFirstName:'',
+            opponentLastName:''
+        }
+        this.briefInfo = {
+            appellant: this.appellant,
+            issuesPresented: "",
+            oralArgumentStatement: "",
+            publicationStatement: "",
+            caseFactsStatement: "",
+            argument: "",
+            conclusion:"",
+            appendexDocuments: ""
+
+        }
+
+    }
+    constructor(private http: Http, private router: Router) { } 
+    onSubmitTemplateBased(form: NgForm) {
+        let params = new URLSearchParams();
+        console.log(form.value);
+        console.log(this.appellant);
+        console.log(this.address);
+        console.log(this.briefInfo);
+        console.log(this.circuitCourtCase);
+        var appJSON = JSON.stringify(this.appellant);
+        console.log(appJSON);
+        var addrJSON = JSON.stringify(this.address);
+        var briefJSON = JSON.stringify(this.briefInfo); 
+        var caseJSON = JSON.stringify(this.circuitCourtCase); 
+        params.append("BriefInfo", briefJSON);
+        params.append("Appellant", appJSON);
+        params.append("Address", addrJSON);
+        params.append("CircuitCourtCase", caseJSON);
+        
+
+        this.http.post('/api/POST', params).subscribe(
+            data => {
+                alert('ok');
+            },
+            error => {
+                console.log(JSON.stringify(error.json()))
+            }
+        )
+        this.router.navigateByUrl('/final');
+        
+        
+    }
+    sendData(info: BriefInfo ) {
+        let params = new URLSearchParams();
+        var sendobj = JSON.stringify(info);
+        params.append("BriefInfo", sendobj);
+        params.append("Appellant", sendobj);
+
+        return this.http.post('/api', params).subscribe(
```

Comment:
Data should be posted to "/api/brief"

Reply:
Didn't know what url to post to before. It's changed!